$OpenBSD: patch-poppler_defs,v 1.3 2015/12/18 22:41:37 kili Exp $

Let this work with poppler >= 0.18.

https://bugs.launchpad.net/poppler-python/+bug/776234

And follow API changes from poppler-0.16.


--- poppler.defs.orig	Sun Sep 27 11:07:59 2009
+++ poppler.defs	Fri Dec 18 13:00:41 2015
@@ -578,18 +578,6 @@
   )
 )
 
-(define-enum Orientation
-  (in-module "Poppler")
-  (c-name "PopplerOrientation")
-  (gtype-id "POPPLER_TYPE_ORIENTATION")
-  (values
-    '("portrait" "POPPLER_ORIENTATION_PORTRAIT")
-    '("landscape" "POPPLER_ORIENTATION_LANDSCAPE")
-    '("upsidedown" "POPPLER_ORIENTATION_UPSIDEDOWN")
-    '("seascape" "POPPLER_ORIENTATION_SEASCAPE")
-  )
-)
-
 (define-enum PageTransitionType
   (in-module "Poppler")
   (c-name "PopplerPageTransitionType")
@@ -1341,11 +1329,6 @@
   (return-type "GType")
 )
 
-(define-function poppler_orientation_get_type
-  (c-name "poppler_orientation_get_type")
-  (return-type "GType")
-)
-
 (define-function poppler_page_transition_type_get_type
   (c-name "poppler_page_transition_type_get_type")
   (return-type "GType")
@@ -1646,58 +1629,6 @@
   (return-type "GType")
 )
 
-(define-method render_to_pixbuf
-  (of-object "PopplerPage")
-  (c-name "poppler_page_render_to_pixbuf")
-  (return-type "none")
-  (parameters
-    '("int" "src_x")
-    '("int" "src_y")
-    '("int" "src_width")
-    '("int" "src_height")
-    '("double" "scale")
-    '("int" "rotation")
-    '("GdkPixbuf*" "pixbuf")
-  )
-)
-
-(define-method render_to_pixbuf_for_printing
-  (of-object "PopplerPage")
-  (c-name "poppler_page_render_to_pixbuf_for_printing")
-  (return-type "none")
-  (parameters
-    '("int" "src_x")
-    '("int" "src_y")
-    '("int" "src_width")
-    '("int" "src_height")
-    '("double" "scale")
-    '("int" "rotation")
-    '("GdkPixbuf*" "pixbuf")
-  )
-)
-
-(define-method get_thumbnail_pixbuf
-  (of-object "PopplerPage")
-  (c-name "poppler_page_get_thumbnail_pixbuf")
-  (return-type "GdkPixbuf*")
-)
-
-(define-method render_selection_to_pixbuf
-  (of-object "PopplerPage")
-  (c-name "poppler_page_render_selection_to_pixbuf")
-  (return-type "none")
-  (parameters
-    '("gdouble" "scale")
-    '("int" "rotation")
-    '("GdkPixbuf*" "pixbuf")
-    '("PopplerRectangle*" "selection")
-    '("PopplerRectangle*" "old_selection")
-    '("PopplerSelectionStyle" "style")
-    '("GdkColor*" "glyph_color")
-    '("GdkColor*" "background_color")
-  )
-)
-
 (define-method render
   (of-object "PopplerPage")
   (c-name "poppler_page_render")
@@ -1795,6 +1726,12 @@
 (define-method get_text
   (of-object "PopplerPage")
   (c-name "poppler_page_get_text")
+  (return-type "char*")
+)
+
+(define-method get_selected_text
+  (of-object "PopplerPage")
+  (c-name "poppler_page_get_selected_text")
   (return-type "char*")
   (parameters
     '("PopplerSelectionStyle" "style")
